gensim/models/poincare.py

```diff
@@ -0,0 +1,637 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Author: Jayant Jain <jayantjain1992@gmail.com>
+# Copyright (C) 2017 Radim Rehurek <me@radimrehurek.com>
+# Licensed under the GNU LGPL v2.1 - http://www.gnu.org/licenses/lgpl.html
+
+
+"""
+Python implementation of Poincare Embeddings, an embedding to capture hierarchical information,
+described in [1]
+
+This module allows training a Poincare Embedding from a training file containing relations from
+a transitive closure.
+
```

Comment:
Code example would be helpful.

Reply:
Short description of "what's the main idea, what's killer-feature of this representation" will be helpful too.